tests/Microsoft.Bot.Builder.Tests/TestHttpClient.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
```

Comment:
// (C)opyright

Reply:
Oh of course. Good catch.

Reply:
This class isn't actually needed ... you can pass an HttpClientHandler into the normal HttpClient, and mock the result there.

Something like:
```cs
            Func<Task<HttpResponseMessage>> createResponseMessage = async () =>
            {
                var response = new HttpResponseMessage(System.Net.HttpStatusCode.OK);
                response.Content = new StringContent(new JObject { { ActivityIdName, ActivityIdValue }, { ConversationIdName, ConversationIdValue } }.ToString());
                return response;
            };

            var mockCredentialProvider = new Mock<ICredentialProvider>();
            var mockHttpMessageHandler = new Mock<HttpMessageHandler>();
            mockHttpMessageHandler.Protected()
                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
                .Returns((HttpRequestMessage request, CancellationToken cancellationToken) => createResponseMessage());

            var httpClient = new HttpClient(mockHttpMessageHandler.Object);
```

Reply:
I may not need to even do that depending on how I end up resolving John's feedback